modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationQueueEntryResourceImpl.java

```diff
@@ -177,8 +177,15 @@ private NotificationQueueEntry _toNotificationQueueEntry(
 				sentDate = serviceBuilderNotificationQueueEntry.getSentDate();
 				status = serviceBuilderNotificationQueueEntry.getStatus();
 				subject = serviceBuilderNotificationQueueEntry.getSubject();
-				triggerBy = notificationHandler.getTriggerBy(
-					contextAcceptLanguage.getPreferredLocale());
+
+				if (notificationHandler != null) {
+					triggerBy = notificationHandler.getTriggerBy(
+						contextAcceptLanguage.getPreferredLocale());
+				}
+				else {
+					triggerBy = "Not Found";
+				}
+
```

Comment:
@devpedrotavares Could you explain how this change fixes the described issue [LPS-175996](https://issues.liferay.com/browse/LPS-175996)? It is expected to return `Not Found`?

In addition, shouldn't you use the language key instead?  `_language.get(locale, "not-found")`, for example

Reply:
Yes, @carolmariaabb. When the objectDefinition that triggered the notification entry is deleted, there will be no notificationHandler for that objectDefinition, which causes NullPointerException, returning 500 status code. That's why doing that null check solves the issue.
With this in mind, I see two alternatives: pass an empty string or pass some message denoting that the trigger object was not found (or does not exist).
If we opt to pass a message, yes we should use the language key as you said, not using it was a mistake.

Reply:
This is the output: 
![not-found-result](https://user-images.githubusercontent.com/82452840/227017030-be3d632c-2f80-4ce0-8d0a-3c9d3c40bb1f.png)

Reply:
@devpedrotavares please confirm this new behavior with the designers and Ana (or Luiz). Thanks!

Reply:
Just updated with the message as discussed in this [thread](https://liferay.slack.com/archives/C026UR5976G/p1679589546371649)